main/static/script.js

```diff
@@ -1,13 +1,17 @@
 "use strict";
 
-console.log("Hello, world!");
-
 var module = angular.module("main",[]);
 
+var chartsReady = false;
+
+var debounce = false;
```

Comment:
When `onSubmitQuery` is called, it changes the hash. So the `hashchange` event fires, which calls `onSubmitQuery` again; both calls to `onSubmitQuery` fetch from the database.

So the `debounce` tells `hashchange` event listener not to call `onSubmitQuery` so that this second, extraneous call doesn't occur.